src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
The multi interface still uses callbacks - so the write callback still needs to block till there is a reader and libcurl does not provide support for that. The only advantage is that curl_mutli_perform does not block but still some sort of loop is needed to check the return value of curl_multi_perform.